api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +561,120 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+
+    private final String crs;
+
+    private GeometryType(String crs) {
+      this.crs = crs;
```

Comment:
Precondition that crs is not null?

Reply:
I think it would be easier to allow null to indicate the default. That's easier to detect for `toString` to show `geometry` instead of `geometry()`.

Reply:
We allow `null` and not allow empty string for CRS. Additionally we check that the CRS should not have leading or trailing spaces to make sure that the value of CRS will be persisted after converting to string and parsing it back.

Reply:
Curious, why not always have toString show the default value like Geography?  Geometry("OGC:CRS84").  Is it to save some bytes ?  Seems to be easier imo.

Reply:
Geography does not always show the default CRS. It happens when the edge algorithm is not using default value but CRS is left unset. We have to fill in the default CRS as the first argument in this case.

I thought that the behavior of toString is to omit optional parameters when possible. See https://github.com/apache/iceberg/pull/12346/files/0c451cac53f4098680aa0ae518af70dc45efaeb1#r1994430054.